src/lib.rs

```diff
@@ -16,168 +17,256 @@ use std::{
     funcs: Arc<bindings::metal_irconverter>,
 }
 
 impl Drop for IRShaderReflection {
+    #[doc(alias = "IRShaderReflectionDestroy")]
     fn drop(&mut self) {
-        unsafe { self.funcs.IRShaderReflectionDestroy(self.me) }
+        unsafe { self.funcs.IRShaderReflectionDestroy(self.me.as_ptr()) }
     }
 }
 
 impl IRShaderReflection {
-    pub fn new(compiler: &IRCompiler) -> Result<IRShaderReflection, Box<dyn std::error::Error>> {
-        unsafe {
-            let me = compiler.funcs.IRShaderReflectionCreate();
-            Ok(Self {
-                me,
-                funcs: compiler.funcs.clone(),
-            })
+    #[doc(alias = "IRShaderReflectionCreate")]
+    pub fn new(compiler: &IRCompiler) -> IRShaderReflection {
+        let me = NonNull::new(unsafe { compiler.funcs.IRShaderReflectionCreate() })
+            .expect("Failed to create IRShaderReflection");
+        Self {
+            me,
+            funcs: compiler.funcs.clone(),
         }
     }
 
-    pub fn get_compute_info(
+    #[doc(alias = "IRShaderReflectionCopyComputeInfo")]
+    pub fn compute_info(
         &self,
         version: ffi::IRReflectionVersion,
-    ) -> Result<ffi::IRVersionedCSInfo, Box<dyn std::error::Error>> {
+    ) -> Option<ffi::IRVersionedCSInfo> {
         let mut info = MaybeUninit::uninit();
         if unsafe {
-            self.funcs
-                .IRShaderReflectionCopyComputeInfo(self.me, version, info.as_mut_ptr())
+            self.funcs.IRShaderReflectionCopyComputeInfo(
+                self.me.as_ptr(),
+                version,
+                info.as_mut_ptr(),
+            )
         } {
-            Ok(unsafe { info.assume_init() })
+            Some(unsafe { info.assume_init() })
         } else {
-            panic!("Test me");
+            None
         }
     }
 }
 
+#[derive(Error, Debug)]
+#[error("Failed to get MetalLib bytecode from IRObject")]
+pub struct MetalLibNoBytecodeFoundError(ffi::IRObjectType, ffi::IRShaderStage);
+
 pub struct IRObject {
-    me: *mut bindings::IRObject,
+    me: NonNull<bindings::IRObject>,
     funcs: Arc<bindings::metal_irconverter>,
 }
 
 impl Drop for IRObject {
+    #[doc(alias = "IRObjectDestroy")]
     fn drop(&mut self) {
-        unsafe { self.funcs.IRObjectDestroy(self.me) }
+        unsafe { self.funcs.IRObjectDestroy(self.me.as_ptr()) }
     }
 }
 
 impl IRObject {
-    pub fn create_from_dxil(
-        compiler: &IRCompiler,
-        bytecode: &[u8],
-    ) -> Result<IRObject, Box<dyn std::error::Error>> {
+    #[doc(alias = "IRObjectCreateFromDXIL")]
+    pub fn create_from_dxil(compiler: &IRCompiler, bytecode: &[u8]) -> IRObject {
         unsafe {
-            let me = compiler.funcs.IRObjectCreateFromDXIL(
+            let me = NonNull::new(compiler.funcs.IRObjectCreateFromDXIL(
                 bytecode.as_ptr(),
                 bytecode.len(),
-                bindings::IRBytecodeOwnership::None,
-            );
+                bindings::IRBytecodeOwnership::Copy,
```

Comment:
Maybe we should document why this is changing (with a comment what `Copy` means). Alternatively we could return an `IRObject<'data>` with `_marker: PhantomData<&'data [u8]>` :)

Reply:
About the PhantomData: if I understand this correctly, we can use that to keep track of the lifetime of the bytecode data slice, by adding an empty dummy field to the struct, which means we won't have to copy the bytecode? Just asking this to learn